pdns/dnsdist.cc

```diff
@@ -721,6 +723,48 @@ shared_ptr<DownstreamState> whashed(const NumberedServerVector& servers, const D
   return valrandom(dq->qname->hash(g_hashperturb), servers, dq);
 }
 
+/*
+ * @todo
+ * - test/benchmark other hashing methods
+ * - test/benchmark adding an avalanche algorithm on hashes
+ * @see https://github.com/haproxy/haproxy/blob/master/doc/internals/hashing.txt
+ */
+
+shared_ptr<DownstreamState> chashed(const NumberedServerVector& servers, const DNSQuestion* dq)
+{
+  std::map<unsigned int, shared_ptr<DownstreamState>> circle = {};
+  unsigned int qhash = dq->qname->hash(g_hashperturb);
+
+  for (const auto& d: servers) {
+    if (d.second->isUp()) {
+      if (d.second->hashes.empty()) {
```

Comment:
We probably need a per-server `mutex` when accessing `hashes`.

Reply:
Definitely !